chandra_aca/drift.py

```diff
@@ -173,3 +174,42 @@ def get_aca_offsets(detector, chip_id, chipx, chipy, time, t_ccd):
     ddz = dz_chip - dz_pred
 
     return ddy, ddz
+
+
+def get_web_zero_offset_table():
+    """
+    Get official SOT MP zero offset aimpoint table:
+
+    :returns: zero offset aimpoint table as astropy.Table
+    """
+    ZOA = "https://icxc.harvard.edu/mp/html/aimpoint_table/zero_offset_aimpoints.txt"
+    page = requests.get(ZOA)
+    table_lines = page.content.strip().split("\n")
+    return Table.read(table_lines, format="ascii")
+
+
+def get_target_aimpoint(date, cycle, detector, too=False, zero_offset_table=None):
+    """
+    Given date, proposal cycle, and detector, return aimpoint chipx, chipy, chip_id
+
+    :param date: observation date
+    :param cycle: proposal cycle of observation
+    :param detector: target detector
+    :param too: boolean. If target is TOO use current cycle not proposal cycle.
+
+    :returns: astropy table row of chipx, chipy, chip_id
+    """
+    if zero_offset_table is None:
+        zero_offset_table = get_web_zero_offset_table()
+    # Get entries for this detector before the 'date' given
+    filtered_table = zero_offset_table[
```

Comment:
Don't put complicated boolean expressions inside a table filter.  Try to find commonalities where you can, in this case the `too` dependent filtering can be done in a common framework with the other filtering.  Use:
```
date = DateTime(date).date[:10]
ok = (detector == zero_offset_table['detector']) & (date >= zero_offset_table['date'])
if not too:
    ok = ok & (cycle >= zero_offset_table['cycle'])

filtered_table = zero_offset_table[ok]
```